cmd/kaspaminer/mineloop.go

```diff
@@ -117,6 +117,10 @@ func handleFoundBlock(client *minerClient, block *externalapi.DomainBlock) error
 
 	rejectReason, err := client.SubmitBlock(block)
 	if err != nil {
+		if nativeerrors.Is(err, router.ErrTimeout) {
+			log.Warnf("Got timeout while submitting block %s to %s: %s", blockHash, client.Address(), err)
+			return nil
```

Comment:
The result of this is that the block will be discarded.
But there's no easy way to solve this right now.
I feel like we should re-write the cpu miner in an async way, that way a separate go-routine will loop trying to submit